src/plugins/data/public/ui/search_bar/create_search_bar.tsx

```diff
@@ -52,73 +56,280 @@ const defaultOnRefreshChange = (query: QueryStart) => {
+    if (props.onSavedQueryIdChange) props.onSavedQueryIdChange(savedQuery.id);
+  };
+};
+
+const defaultOnQuerySaved = (
+  props: StatefulSearchBarProps,
+  queryService: QueryStart,
+  setQueryStringState: Function,
+  setSavedQueryState: Function
+) => {
+  if (!props.useDefaultBehaviors) return props.onSaved;
+  return (savedQuery: SavedQuery) => {
+    populateStateFromSavedQuery(
+      props,
+      queryService,
+      savedQuery,
+      setQueryStringState,
+      setSavedQueryState
+    );
+    if (props.onSavedQueryIdChange) props.onSavedQueryIdChange(savedQuery.id);
+  };
+};
+
+const overrideDefaultBehaviors = (props: StatefulSearchBarProps) => {
+  return props.useDefaultBehaviors ? {} : props;
+};
+
 export function createSearchBar({ core, storage, data }: StatefulSearchBarDeps) {
   // App name should come from the core application service.
   // Until it's available, we'll ask the user to provide it for the pre-wired component.
   return (props: StatefulSearchBarProps) => {
-    const { filterManager, timefilter } = data.query;
+    const { appName, savedQueryId, onQuerySubmit } = props;
+    const { filterManager, timefilter, savedQueries } = data.query;
     const tfRefreshInterval = timefilter.timefilter.getRefreshInterval();
-    const fmFilters = filterManager.getFilters();
     const [refreshInterval, setRefreshInterval] = useState(tfRefreshInterval.value);
     const [refreshPaused, setRefreshPaused] = useState(tfRefreshInterval.pause);
+    const [timeRange, setTimerange] = useState(timefilter.timefilter.getTime());
+
+    // Handle queries
+    const [query, setQuery] = useState<Query>(
+      props.query || {
+        query: '',
+        language: core.uiSettings.get('search:queryLanguage'),
+      }
+    );
 
-    const [filters, setFilters] = useState(fmFilters);
+    // Fire onQuerySubmit on query or timerange change
+    useEffect(() => {
+      if (onQuerySubmit)
+        onQuerySubmit(
+          {
+            dateRange: timeRange,
+            query,
+          },
+          true
+        );
+    }, [onQuerySubmit, query, timeRange]);
 
-    // We do not really need to keep track of the time
-    // since this is just for initialization
-    const timeRange = timefilter.timefilter.getTime();
+    // Handle saved queries
+    const [savedQuery, setSavedQuery] = useState<SavedQuery>();
+    useEffect(() => {
+      const fetchSavedQuery = async () => {
+        if (savedQueryId) {
+          try {
+            const newSavedQuery = await savedQueries.getSavedQuery(savedQueryId);
+            // Make sure we set the saved query to the most recent one
+            if (newSavedQuery && newSavedQuery.id === savedQueryId) {
+              setSavedQuery(newSavedQuery);
+            }
+          } catch (error) {
+            // Clear saved query
+            setSavedQuery(undefined);
+            core.notifications.toasts.addWarning({
+              title: i18n.translate('xpack.maps.unableToGetSavedQueryToastTitle', {
```

Comment:
Current i18n has maps in path `xpack.maps.unableToGetSavedQueryToastTitle`. Maybe change this to something else?

Reply:
done